### PR TITLE
Abbreviate nested enums

### PR DESCRIPTION
**Pull request by stephencelis**
We can use dot-abbreviation to keep descriptions short but provide enough context.

### CODE REVIEW

File: Sources/CustomDump/Dump.swift [hunk 1]
```diff
@@ -58,14 +58,15 @@ public func customDump<T, TargetStream>(
 ) -> T where TargetStream: TextOutputStream {
 
   var idPerItem: [ObjectIdentifier: UInt] = [:]
-  var occurencePerType: [String: UInt] = [:]
+  var occurrencePerType: [String: UInt] = [:]
```
Comment (stephencelis) [inline Sources/CustomDump/Dump.swift:61]: Unrelated typo fix.

**Review by mbrandonw — APPROVED** [review]
